core/chains/evm/logpoller/orm.go

```diff
@@ -272,7 +272,7 @@ func (o *DbORM) DeleteExpiredLogs(limit int64, qopts ...pg.QOpt) (int64, error)
 				GROUP BY evm_chain_id, address, event
 				HAVING NOT 0 = ANY(ARRAY_AGG(retention))
 			) r ON l.evm_chain_id = $1 AND l.address = r.address AND l.event_sig = r.event
-			AND l.block_timestamp <= STATEMENT_TIMESTAMP() - (r.retention / 10^9 * interval '1 second')
```

Comment:
Curious to hear your reasoning for the block_timestamp -> created_at change.

I was thinking block_timestamp makes more sense, because how long we retain a log seems like it should be based on when the log was emitted rather than when we happened to find it. For example, if a log is a week old and the retention is 24 hours, but the node happens to have been down for a week... when we spin up the node it would backfill that log and then retain it for another 24 hours.  Seems like it ought to just get rid of it immediately since it's been way more than 24 hours since it was emitted.

Reply:
That actually makes me think of another optimization I should add:  if a log comes back from the rpc server and it's already past its retention time we shouldn't save it to the db in the first place; I don't think that check is in there yet, but it should be.

Reply:
At first i wanted that to use block_timestamp https://github.com/smartcontractkit/chainlink/pull/12060, but I'm not sure about replaying logs older than retention, they will be immediately wiped out from db